syntax = "proto3";

package envoy.config.filter.network.trpc_proxy.v3;

import "envoy/config/route/v3/route.proto";
import "envoy/config/accesslog/v3/accesslog.proto";
import "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto";
import "validate/validate.proto";

message TrpcProxy {
    oneof route_specifier {
        option (validate.required) = true;

        extensions.filters.network.http_connection_manager.v3.Rds rds = 1;

        config.route.v3.RouteConfiguration route_config = 2;
    }

    repeated config.accesslog.v3.AccessLog access_log = 3;

    string stat_prefix = 4; // 出统计信息时使用的一个前缀
}